OutOfSchool/OutOfSchool.WebApi/Extensions/Startup/QuartzExtension.cs

```diff
@@ -0,0 +1,60 @@
+using System;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using MySqlConnector;
+using OutOfSchool.Common.Extensions.Startup;
+using OutOfSchool.WebApi.Config;
+using OutOfSchool.WebApi.Services.Elasticsearch;
+using OutOfSchool.WebApi.Util;
+using Quartz;
+
+namespace OutOfSchool.WebApi.Extensions.Startup
+{
+    public static class QuartzExtension
+    {
+        public static IServiceCollection AddDefaultQuartz(
+            this IServiceCollection services,
+            IConfiguration configuration)
+        {
+            _ = services ?? throw new ArgumentNullException(nameof(services));
+
+            services.AddQuartz(q =>
+            {
+                q.SchedulerId = "DefaultAppQuartz";
+                q.SchedulerName = "DefaultAppQuartz";
+                q.SetProperty("quartz.serializer.type", "json");
```

Comment:
It seems you can read all the quartz configuration from the external config and use `SetProperty` for each.
It will bring mode flexibility to the server administrator, but it should be well documented then.
In case you want to restrict user to config everything - a way better to have a particular config model.

Reply:
rewrote this